Microsoft.Toolkit.Uwp.UI.Controls/TextBoxRegex/TextBoxRegex.Properties.cs

```diff
@@ -0,0 +1,126 @@
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// TextBoxRegex allows text validation using a regular expression.
+    /// </summary>
+    /// <remarks>
+    /// If<see cref="ValidationMode"> is set to Normal then IsValid will be set according to either the regex is valid.</see>
+    /// If<see cref="ValidationMode"> is set to Forced and the input is not valid the TextBox text will be cleared.</see>
+    /// </remarks>
+    public partial class TextBoxRegex
+    {
+        /// <summary>
+        /// The regex expression that will be validated on the textbox
+        /// </summary>
+        public static readonly DependencyProperty RegexProperty = DependencyProperty.RegisterAttached("Regex", typeof(string), typeof(TextBoxRegex), new PropertyMetadata(null, TextBoxRegexPropertyOnChange));
+
+        /// <summary>
+        /// Gets the result of the TextBoxRegex validation again the Regex property
+        /// </summary>
+        public static readonly DependencyProperty IsValidProperty = DependencyProperty.RegisterAttached("IsValid", typeof(bool), typeof(TextBoxRegex), new PropertyMetadata(false));
+
+        /// <summary>
+        /// The validation mode of the validation extension (Normal, Forced)
+        /// </summary>
+        public static readonly DependencyProperty ValidationModeProperty = DependencyProperty.RegisterAttached("ValidationMode", typeof(ValidationMode), typeof(TextBoxRegex), new PropertyMetadata(ValidationMode.Normal, TextBoxRegexPropertyOnChange));
+
+        /// <summary>
+        /// The validation types are predefined validation that can be used directly
+        /// </summary>
+        public static readonly DependencyProperty ValidationTypeProperty = DependencyProperty.RegisterAttached("ValidationType", typeof(ValidationType), typeof(TextBoxRegex), new PropertyMetadata(ValidationType.Custom, TextBoxRegexPropertyOnChange));
+
+        /// <summary>
+        /// Gets the Regex property
+        /// </summary>
+        /// <param name="obj">TextBox to get Regex property from.</param>
+        /// <returns>The regular expression assigned to the TextBox</returns>
+        public static string GetRegex(DependencyObject obj)
+        {
+            return (string)obj.GetValue(RegexProperty);
+        }
+
+        /// <summary>
+        /// Set Regex property
```

Comment:
Sets the value of the TextBoxRegEx.RegEx XAML attached property for a target TextBox.

Following pattern set by framework team